parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java

```diff
@@ -74,19 +78,28 @@ public static WriterVersion fromString(String name) {
   private final int dictionaryPageSizeThreshold;
   private final WriterVersion writerVersion;
   private final boolean enableDictionary;
+  // TODO - Should this even hold an allocator? this isn't really a property
+  private ByteBufferAllocator allocator;
 
   public ParquetProperties(int dictPageSize, WriterVersion writerVersion, boolean enableDict) {
+    this(dictPageSize, writerVersion, enableDict, new HeapByteBufferAllocator());
+  }
+
+  public ParquetProperties(int dictPageSize, WriterVersion writerVersion, boolean enableDict, ByteBufferAllocator allocator) {
```

Comment:
If we are going to go with any pattern, I think it would be best to just go completely with a builder, this would allow for us to enforce required/optional parameters properly. For us to do this in the pattern you suggest we would have to have a validateSetup() method as well which would have to be called after setting all of the properties. Calling this method could not be enforced by the compiler, which would make it a little brittle going forward.